be/src/agent/heartbeat_server.cpp

```diff
@@ -147,6 +147,10 @@ Status HeartbeatServer::_heartbeat(
         _master_info->__set_http_port(master_info.http_port);
     }
 
+    if (master_info.__isset.heartbeat_flag) {
+        _olap_engine->set_heartbeat_flag(master_info.heartbeat_flag);
```

Comment:
I prefer that we store this flag in exec_env, class who has the requirement access the flag it needs.
If changing in your way, there is no place to see what's the flags is. And what's more, if someone needs this flags, it will be set here, which is high coupling.

Reply:
done